File: app/src/main/java/com/returnnotfound/stain/di/module/RepositoryModule.java
package com.returnnotfound.stain.di.module;

import com.returnnotfound.stain.data.repository.AnimeRepository;
import com.returnnotfound.stain.data.repository.AnimeRepositoryImpl;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class RepositoryModule {

  @Binds
  abstract AnimeRepository bindAnimeRepository(AnimeRepositoryImpl animeRepository);
}

File: app/src/main/java/com/returnnotfound/stain/di/module/RepositoryModule.kt
package com.returnnotfound.stain.di.module

import com.returnnotfound.stain.data.repository.BannerRepositoryImpl
import com.returnnotfound.stain.domain.repository.BannerRepository
import dagger.Binds
import dagger.Module

@Module
abstract class RepositoryModule {
  @Binds
  abstract fun bindBannerRepository(bannerRepositoryImpl: BannerRepositoryImpl): BannerRepository
}